### PR TITLE
React capstone project

### PR DESCRIPTION
**Pull request by dfsalam**
### Milestone 1: Setup and show data
- Added react boilerplate.
- Setup file structure.
- Added app skeleton.
- Displayed towns as buttons.
- Showed pollution data in a separate route for each town.
### Milestone 2: Add search component
- In this pull request I implemented the search component and added the reducer searchTown inside townsSlice for the functionality.
### Milestone 3: Styling
- Fixed error on details page load.
- Added style for all components.
- Deleted console.logs and fix linter errors.
### Milestone 4: Testing
- In this pull request I added tests for all components and for the reducers.
### Milestone 5: Readme and additional test.
- In this pull request I added the readme file and some additional tests.

### CODE REVIEW

**Review by KDlamini — CHANGES_REQUESTED** [review]
### Hi @dfsalam 🙋‍♀️,  

Great work so far! 💯 
You have done a great job creating a working metrics app using React and redux and following the [Behance design specifications](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)). 🎊 
There are some issues that you still need to work on improving your project according to the requirements. You are almost there! 💪🏾 

## Highlights 🌞 

✔️  Redux state management and data retrieved from the API are stored in the Redux store correctly
✔️  The Home page has a list of categories that could be filtered using given parameters
✔️  It's possible to navigate the detail page by clicking the individual category item
✔️ The category item detail page has its unique route within the SPA
✔️ All unit tests pass and sufficient tests are written

## Video Presentation 📹 

✔️  There's a recorded video presenting the features of the project with a  maximum length of 5 minutes.
✔️ The video contains a clear and objective description of the project in professional English and the topic and purpose of the project are made clear.
✔️ The video contains a demo of the project, highlighting all the project features.
✔️ The video contains a highlight about a piece of the code the student finds interesting or is proud of making.

## Required Changes ♻️

- Please make sure that the Pull Request description is more detailed about the introduced changes.

For example:
    - The project is a single-page application (SPA) built with React and Redux.
    - The data retrieved from the [API you used] and stored in the Redux store.
    - The is a search filter for the data retrieved from the API using a Filter stateless component.
    - Every page (the main page and the pages for each item) have a unique route within the SPA using `react-router-dom`.
    - The project is deployed and accessible online using [GitHub Pages](https://pages.github.com/).

#### Home page
-  On page loads, the web app shows the list of categories that could be filtered out by category name.
- Each category displays some numeric values per category that come from the API.
- When the user clicks (or taps) on a category item, the application navigates to the details page.

#### Details page

- On the details page, the web app retrieves data from the API to show detailed data in the category.
- When the user clicks on the "Back" button (<), the user navigates to the home page.

 
**Kindly look through the comments below 👇 for required changes**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/routes/Home.jsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    );
+  }
+
+  return (
+    <div>
+      <Search />
+      <TownContainer />
+    </div>
+  );
+};
```
[the remark this answers is not in the document]
Comment (dfsalam) [reply src/routes/Home.jsx]: Thank you. Done.

File: src/components/Detail.jsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  const { polution } = useAppSelector((state) => state.polution);
+  if (polution.length === 0) {
+    return (<h3>Is loading...</h3>);
+  }
+  const { lat, lon } = polution.coord;
+  const {
+    co, no, no2, o3, pm25, pm10, so2,
+  } = polution.list[0].components;
+  return (
+    <div className="detail-container">
```
[the remark this answers is not in the document]
Comment (dfsalam) [reply src/components/Detail.jsx]: I agree. Done.

File: src/components/Search.jsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { useAppDispatch } from '../redux/redux-hooks';
+import { searchTown } from '../redux/towns/townsSlice';
+import '../styles/Search.css';
+
+const Search = () => {
+  const dispatch = useAppDispatch();
+  const handler = (desired) => {
```
[the remark this answers is not in the document]
Comment (dfsalam) [reply src/components/Search.jsx:8]: Thank you. Done.

**Review by muneebulrehman — APPROVED** [review]
Hi @dfsalam , 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🥳
Congratulations! 🎉 

# Status: Approved✔️

## Highlights
- No linter errors.
- Descriptive PR.
- Good follow-up with the previous reviewer.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**